docs/docs/api-ref.md

```diff
@@ -0,0 +1,261 @@
+
+### Create Site
+
+Creates a new site for the given site item object.
+
+```py
+Sites.create(new_site_item)
+```
+
+Example:
+
+```py
+new_site = TSC.SiteItem(name='Tableau', content_url='tableau', admin_mode=TSC.SiteItem.AdminMode.ContentAndUsers, user_quota=15, storage_quota=1000, disable_subscriptions=True)
+self.server.sites.create(new_site)
+```
+
+### Get Site by ID
+
+Gets the site with the given ID.
+
+```py
+Sites.get_by_id(id)
+```
+
+### Get Sites
+
+Gets all the sites on the server.
+
+```py
+Sites.get()
+```
+
+### Update Site
+
+Modifies a site. The site item object must include the site ID and overrides all other settings.
+
+```py
+Sites.update(site_item_object)
+```
+
+### Delete Site
+
+Deletes the site with the given ID.
+
+```py
+Sites.delete(id)
+```
+
+## Projects
+
+Source files: server/endpoint/projects_endpoint.py
+
+### Create Project
+
+Creates a project for the given project item object.
+
+```py
+Projects.create(project_item_object)
+```
+
+Example:
+
+```py
+new_project = TSC.ProjectItem(name='Test Project', description='Project created for testing')
+new_project.content_permissions = 'ManagedByOwner'
+self.server.projects.create(new_project)
+```
+
+### Get Projects
+
+Get a list of projects on the server.
+
+```py
+Projects.get()
+```
+
+### Update Project
+
+Modifies a project. The project item object must include the project ID and overrides all other settings.
+
+```py
+Projects.update(project_item_object)
+```
+
+### Delete Project
+
+Deletes a project by ID.
+
+```py
+Projects.delete(id)
+```
+
+## Workbooks
+
+Source files: server/endpoint/workbooks.py, models/workbook_item.py
+
+### Get Workbooks
+
+Get all workbooks on the server.
+
+```py
+Workbooks.get()
+```
+
+### Get Workbook by ID
+
+Gets a workbook with a given ID.
+
+```py
+Workbooks.get_by_id(id)
+```
+
+### Publish Workbook
+
+Publish a local workbook to Tableau Server.
+
+```py
+Workbooks.publish(workbook_item, file_path, publish_mode)
+```
+
+Where the publish mode is one of the following:
+
+* Append
+* Overwrite
+* CreateNew
+
+Example:
+
+```py
+wb_item = TSC.WorkbookItem(name='Sample',
+                                show_tabs=False,
+                                project_id='ee8c6e70-43b6-11e6-af4f-f7b0d8e20760')
+
+server.workbooks.publish(wb_item,
+                         os.path.join(YOUR_DIR, 'SampleWB.twbx'),
+                         self.server.PublishMode.CreateNew)
+```
+
+### Update Workbook
+
+Modifies a workbook. The workbook item object must include the workbook ID and overrides all other settings.
+
+```py
+Workbooks.update(wb_item_object)
+```
+
+### Delete Workbook
+
+Deletes a workbook with the given ID.
+
+```py
+Workbooks.delete(id)
+```
+
+### Download Workbook
+
+Downloads a workbook to the specified directory.
+
+```py
+Workbooks.download(id, file_path)
+```
+
+### Populate Views for a Workbook
+
+Populates a list of views for a workbook object. You must populate views before you can iterate through the views.
```

Comment:
Populates where? I think this one needs a deeper explanation to really get the point across (sadly 😢 )